app/src/androidTest/java/ch/epfl/polychef/images/ImageStorageTest.java

```diff
@@ -46,9 +52,11 @@ public void initMockFirebaseStorage() {
         MockitoAnnotations.initMocks(this);
         successTask = Tasks.forResult(new byte[] {1, 2, 3});
         failTask = Tasks.forException(new Exception());
+        imageStorageSpy = Mockito.spy(ImageStorage.getInstance());
+        when(imageStorageSpy.getStorage()).thenReturn(firebaseStorage);
```

Comment:
Nice way to mock singleton pattern classes, we may use this elsewhere as well

Reply:
Yeah, I wish we had found it sooner